streams/shared/src/main/scala/zio/stream/ZChannel.scala

```diff
@@ -2021,7 +2023,7 @@ object ZChannel {
                              }
                        }
         _ <- pullStrategy(fiberId).forever
-               .onError(cause =>
```

Comment:
Why not leave it like that and ignore errors? Because that's exactly what I wanted with `catchAllCause` but you demonstrated its surprising behaviour in the face of races

Reply:
There is another key difference between `onError` and `catchAllCause`, which is that in case of interruption `onError` is guaranteed to be invoked, whereas `catchAllCause` won't be unless it's executed within an uninterruptible region.

The cause of the linked issue in this case is that if we use `onError` we're adding the interruption to the queue, whereas if we use `catchAllCause` we're not, which for some reason is causing the issue in zio-kafka